### PR TITLE
feat(RwaUrn): use `file` to control max gem cap

### PR DESCRIPTION
**Pull request by nduchak**

### CODE REVIEW

File: src/RwaUrn.sol [hunk 1]
```diff
@@ -190,7 +189,7 @@ contract RwaUrn2 {
         DaiAbstract(DaiJoinAbstract(daiJoin_).dai()).approve(daiJoin_, type(uint256).max);
         VatAbstract(vat_).hope(daiJoin_);
 
-        emit Recap(gemCap_);
+        emit File("gemCap", gemCap_);
         emit Rely(msg.sender);
         emit File("outputConduit", outputConduit_);
         emit File("jug", jug_);
```
Comment (hbarcelos) [inline src/RwaUrn.sol]: I'd put the `File("gemCap", gemCap_)` last to respect the order they are passed in.

Comment (nduchak) [reply src/RwaUrn.sol]: 👍

Comment (nduchak) [reply src/RwaUrn.sol]: Fixed

File: src/RwaUrn.sol [hunk 2]
```diff
@@ -108,6 +108,12 @@ contract RwaUrn2 {
      * @param data The new value of the parameter.
      */
     event File(bytes32 indexed what, address data);
+    /**
```
Comment (kauenet) [inline src/RwaUrn.sol:75]: I believe this is duplicated.

File: src/RwaUrn.sol [hunk 3]
```diff
@@ -259,14 +258,19 @@ contract RwaUrn2 {
...
-     * @notice Updates the max gem amount this contract can lock.
-     * @param wad Gem amount.
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"gemCap"
+     * @param data The new value of the parameter.
      */
-    function recap(uint256 wad) external auth {
-        require(wad <= 2**255 - 1, "RwaUrn2/overflow");
-        gemCap = wad;
+    function file(bytes32 what, uint256 data) external auth {
```
Comment (kauenet) [inline src/RwaUrn.sol:229]: We could go for something like this:

```
    /**
     * @notice Updates a contract parameter.
     * @param what The changed parameter name. `"outputConduit" | "jug" | "gemCap"`
     * @param data The new value of the parameter.
     */
    function file(bytes32 what, address data) external auth {
        if (what == "outputConduit") {
            require(data != address(0), "RwaUrn2/invalid-conduit");
            outputConduit = data;
        } else if (what == "jug") {
            jug = JugAbstract(data);
        } else if (what == "gemCap") {
            require(data <= 2**255 - 1, "RwaUrn2/overflow");
            gemCap = data;
        } else {
            revert("RwaUrn2/unrecognised-param");
        }

        emit File(what, data);
    }
```

**Pull request merged**